packages/@okta/vuepress-site/docs/guides/migrate-to-oie/index.md

```diff
@@ -0,0 +1,155 @@
+
+```html
+<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
+<html>
+<head>
+    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta name="robots" content="none" />
+
+    <title>{{pageTitle}}</title>
+    {{{SignInWidgetResources}}}
+</head>
+<body>
+    <div class="login-bg-image" style="background-image: {{bgImageUrl}}"></div>
+    <div id="okta-login-container"></div>
+
+    <!--
+        "OktaUtil" defines a global OktaUtil object
+        that contains methods used to complete the Okta login flow.
+     -->
+    {{{OktaUtil}}}
+
+    <script type="text/javascript">
+        // "config" object contains default widget configuration
+        // with any custom overrides defined in your admin settings.
+        var config = OktaUtil.getSignInWidgetConfig();
+
+        // Render the Okta Sign-In Widget
+        var oktaSignIn = new OktaSignIn(config);
+        oktaSignIn.renderEl({ el: '#okta-login-container' },
+            OktaUtil.completeLogin,
+            function(error) {
+                // Logs errors that occur when configuring the widget.
+                // Remove or replace this with your own custom error handler.
+                console.log(error.message, error);
+            }
+        );
+    </script>
+</body>
+</html>
+```
+
+### Embedded signin widget
+
+Set the option `useInteractionCodeFlow` to `true` on the object passed to the Sign-In Widget constructor. This will enable Okta Identity Engine for the widget. Both the authorization server and the application must have the [interaction code](#enable-interaction-code-grant) grant type enabled.
+
+> **Note:** Your code may break if it is calling the `renderEl` method and expects `sessionToken` or `session.setCookieAndRedirect` on the response object. Instead of `renderEl`, we recommend calling the `showSignInToGetTokens` method. This method receives and returns tokens without any browser redirect.
+
+```javascript
+var signIn = new OktaSignIn(
+  {
+    baseUrl: 'https://{yourOktaDomain}',
+    clientId: '{{clientId of your OIDC app}}'
+    redirectUri: '{{redirectUri configured in OIDC app}}'
+    useInteractionCodeFlow: true
+    // other options...
+  }
+);
+
+signIn.showSignInToGetTokens({
+  // Assumes there is an empty element on the page with an id of 'osw-container'
+  el: '#osw-container'
+}).then(function(tokens) {
+  // Store tokens
+  signIn.authClient.tokenManager.setTokens(tokens);
+  // Remove widget
+  signIn.remove();
+  // Can now render in an authenticated state
+}).catch(function(error) {
+  // Handle error
+})
+```
+
+### Direct authentication
+
+Javascript clients, whether browser-based or running in nodeJS, can use the [okta-idx-js](https://github.com/okta/okta-idx-js) and [okta-auth-js](https://github.com/okta/okta-auth-js) SDKs to authenticate using the interaction code flow.
```

Comment:
Have we confirmed server-side works since we moved the PKCE portion to authJS?

I'd prefer to encourage the widget here, since using the idx SDK directly means the dev is on the hook for building a UI based on the responses.

Reply:
yes there are tests on the auth-js side. The developer must polyfill crypto and TextEncoder as we do in our tests. This section is not added as encouragement but as reference.